### DISCUSSION TITLE
Issues with Mac M1 (ARM64)

### DISCUSSION THREAD

**Discussion by joaquincabezas** [LukasZahradnik/PyNeuraLogic#46] [General]
Hi! I've recently migrated my development from a Linux PC to a Mac M1 and found some issues with Java versions. If someone is having issues this post can help.

I was receiving the following error:

`java.lang.reflect.InaccessibleObjectException: java.lang.reflect.InaccessibleObjectException: Unable to make field private final java.lang.String [java.io](http://java.io/).File.path accessible: module java.base does not “opens [java.io](http://java.io/)” to unnamed module`

Another issue was not being able to finish the call to `template.draw()`

From the documentation of PyNeuraLogic, Java 1.8 is recommended, so I did a search on SO finding this answer (https://stackoverflow.com/questions/72769846/will-x64-jdk-1-8-work-in-mac-with-apple-silicon-m1-chip) which proposes using an ARM version provided by Azul (www.azul.com). Using the 8 version was really slow, so I tested with 11 and it is fine!

The working set-up was:

- PyNeuraLogic to the last version (currently 0.7.3)
- Java 11 from Azul (direct link https://www.azul.com/downloads/?version=java-11-lts&os=macos&architecture=arm-64-bit&package=jdk)
- `export JAVA_HOME=/Library/Java/JavaVirtualMachines/zulu-11.jdk/Contents/Home`

Hope this helps other Mac M1 users!

**Comment by LukasZahradnik** [comment]
@joaquincabezas Hello, thanks for providing a solution to this issue.

I came across a similar one recently and managed to fix it. Could you please check if you are getting the error on version 0.7.5 of PyNeuraLogic?

Thanks!

**Reply by joaquincabezas** [reply]
Hi!

With 0.7.5 I am not getting the error :) I can confirm I can execute pyneuralogic code with Oracle JDK 20 and JDK 17 without any specific configuration.

Thanks!!
